packages/node-core/src/indexer/fetch.service.ts

```diff
@@ -194,11 +201,17 @@ export abstract class BaseFetchService<DS extends BaseDataSource, B extends IBlo
    *
    * @param startBlockHeight
    * @param endBlockHeight is either FinalizedHeight or BestHeight, ensure ModuloBlocks not greater than this number
+   * @param moduloNumbers valid modulo number from handler filters
    */
-  getEnqueuedModuloBlocks(startBlockHeight: number, endBlockHeight: number): (IBlock<FB> | number)[] {
+  private getEnqueuedModuloBlocks(
```

Comment:
Why not merge this with `getModulorBlocks` and move a lot of the logic from here https://github.com/subquery/subql/pull/2331/files#diff-04fd5e8c6ab146427c7f38891863dab7467a38756570af71f587358719cd9c71R251-R265 into it too